texttospeech/list_voices/list_voices.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Command list_voices lists the available text to speech voices.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+
+	"golang.org/x/net/context"
+
+	texttospeech "cloud.google.com/go/texttospeech/apiv1"
+	texttospeechpb "google.golang.org/genproto/googleapis/cloud/texttospeech/v1"
+)
+
+// [START tts_list_voices]
+// ListVoices lists the available text to speech voices.
+func ListVoices(out io.Writer) error {
+	ctx := context.Background()
+
+	client, err := texttospeech.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Performs the list voices request.
+	resp, err := client.ListVoices(ctx, &texttospeechpb.ListVoicesRequest{})
+	if err != nil {
+		return err
+	}
+
+	for _, voice := range resp.Voices {
+		// Display the voice's name. Example: tpc-vocoded
+		fmt.Fprintf(out, "Name: %v\n", voice.Name)
+
+		// Display the supported language codes for this voice. Example: "en-US"
+		for languageCode := range voice.LanguageCodes {
+			fmt.Fprintf(out, "  Supported language: %v\n", languageCode)
+		}
+
+		// Display the SSML Voice Gender.
+		fmt.Fprintf(out, "  SSML Voice Gender: %v\n", voice.SsmlGender.String())
+
+		// Display the natural sample rate hertz for this voice. Example: 24000
+		fmt.Fprintf(out, "  Natural Sample Rate Hertz: %v\n",
+			voice.NaturalSampleRateHertz)
+	}
+
+	return nil
+}
+
+// [END tts_list_voices]
+
+func main() {
+	var buf bytes.Buffer
+	err := ListVoices(&buf)
```

Comment:
super nit: you can use `os.Stdout` as the writer here, but the buffer is fine, too.

Reply:
Nice! Didn't know that, changed